src/GraphQL/Validation/Complexity/ComplexityAnalyzer.cs

```diff
@@ -9,11 +10,11 @@ namespace GraphQL.Validation.Complexity
     public class ComplexityAnalyzer : IComplexityAnalyzer
     {
         /// <inheritdoc/>
-        public void Validate(GraphQLDocument document, ComplexityConfiguration complexityParameters)
+        public void Validate(GraphQLDocument document, ComplexityConfiguration complexityParameters, ISchema? schema = null)
         {
             if (complexityParameters == null)
                 return;
-            var complexityResult = Analyze(document, complexityParameters.FieldImpact ?? 2.0f, complexityParameters.MaxRecursionCount);
+            var complexityResult = Analyze(document, complexityParameters.FieldImpact ?? 2.0f, complexityParameters.MaxRecursionCount, schema);
```

Comment:
Why do we have a constant of 2.0f here?  Should it be 2.0f?  It seems like the default impact should be 1.0f.  Either way, perhaps we should convert this to a constant within this class.  For example:

```cs
const float DEFAULT_FIELD_IMPACT = 2.0f
```

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/blob/develop/src/GraphQL/Validation/Complexity/ComplexityAnalyzer.cs#L16

Reply:
@Shane32 Another observation from me - look at `GetImpactFromArgs`. If we have `id` argument then method returns 1 so `CurrentEndNodeImpact` will be less 2 times. Why so? I don't know.